client/components/plans/plan-actions/index.jsx

```diff
@@ -95,57 +101,87 @@ module.exports = React.createClass( {
 			<div>
 				<button
 					className="button is-primary plan-actions__upgrade-button"
-					onClick={ this.handleAddToCart.bind( null, this.cartItem( { isFreeTrial: false } ), 'button' ) }>
+					disabled={ this.props.isSubmitting }
+					onClick={ this.handleSelectPlan }>
 					{ label }
 				</button>
 			</div>
 		);
 	},
 
-	recordStartFreeTrialClick: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Start Free Trial Button', 'Product ID', cartItem.product_id );
+	recordStartFreeTrialClick: function() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Start Free Trial Button', 'Product ID', this.props.plan.product_id );
 	},
 
-	recordUpgradeNowClick: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Link', 'Product ID', cartItem.product_id );
+	recordUpgradeNowClick: function() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Link', 'Product ID', this.props.plan.product_id );
 	},
 
-	recordUpgradeNowButton: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Button', 'Product ID', cartItem.product_id );
+	recordUpgradeNowButton: function() {
+		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Button', 'Product ID', this.props.plan.product_id );
 	},
 
-	recordUpgradeTrialNowClick: function( cartItem ) {
-		analytics.ga.recordEvent( 'Upgrades', 'Clicked Upgrade Now Link For Trial', 'Product ID', cartItem.product_id );
+	getCartItem: function( properties ) {
+		return cartItems.getItemForPlan( this.props.plan, properties );
 	},
 
-	handleAddToCart: function( cartItem, actionType, event ) {
+	handleSelectPlan: function( event ) {
 		if ( event ) {
 			event.preventDefault();
 		}
 
-		if ( cartItem && actionType ) {
-			if ( actionType === 'button' ) {
-				if ( cartItem.free_trial ) {
-					this.recordStartFreeTrialClick( cartItem );
-				}
-				if ( ! cartItem.free_trial ) {
-					this.recordUpgradeNowButton( cartItem );
-				}
-			}
+		if ( this.props.isSubmitting ) {
+			return;
+		}
 
-			if ( actionType === 'link' ) {
-				if ( cartItem.free_trial ) {
-					this.recordUpgradeTrialNowClick( cartItem );
-				}
-				if ( ! cartItem.free_trial ) {
-					this.recordUpgradeNowClick( cartItem );
-				}
-			}
+		const actionType = event.target.tagName === 'A' ? 'link' : 'button';
```

Comment:
I think it's fine for the moment, a click event will always have a target prop and an HTML element will always have a tagName, so it should not break. I don't think it's really fragile per say but I agree that it's not easily extensible.